common/amount.c

```diff
@@ -421,6 +421,10 @@ struct amount_sat amount_tx_fee(u32 fee_per_kw, size_t weight)
 {
 	struct amount_sat fee;
 
+	/* Handle zero case */
+	if (fee_per_kw == 0)
+		return AMOUNT_SAT(0);
+
```

Comment:
True, but the rest of the code is very cheap, so this seems a weird case to optimize?

http://www2.ing.unipi.it/~a009435/issw/extra/kp_elems_of_pgmng_sty.pdf Page 131.